### DISCUSSION TITLE
Custom fields and switching payment method does not keep data in custom fields

### DISCUSSION THREAD

**Discussion by jobst** [impress-org/givewp#7139] [Developer Q&A]
I have followed the following guidelines to add custom fields:

1.  https://github.com/impress-org/givewp-snippet-library/blob/master/donors/add-donor-meta.php
2.  https://github.com/impress-org/givewp-snippet-library/blob/master/form-customizations/custom-fields-plugin.php

I have no problems getting the data into the DB and displaying the data in the various GIVE forms.

**Problem**
After entering data in a custom field in a CLASSIC form then switching payment methods the data is kept in the standard fields (e.g. name) but NOT in the custom fields.

**Steps to reproduce**

1. follow the guidelines in https://github.com/impress-org/givewp-snippet-library/blob/master/donors/add-donor-meta.php
2. launch a CLASSIC form
3. enter data into all fields including custom fields
4. data is kept in standard fields
5. data is deleted in custom fields

**Attached files**

- image_before: shows data in fields including phone BEFORE switching payment method
- image_after: shows data in fields but phone is deleted AFTER switching payment method

Have I fogotten to add code to have the custom fields NOT disappear?

**Attached images follow**

![image_after](https://github.com/impress-org/givewp/assets/1762694/f3870943-1ea8-4ac3-9e5c-39512c9371c8)
![image_before](https://github.com/impress-org/givewp/assets/1762694/d0693ac3-b6cc-4fc5-aaab-691d81f39ec4)

**Comment by kjohnson** [comment]
This sounds like a bug in the v2 codebase with how data is managed when parts of the form are re-rendered.

I've added this report to our feedback system so that the right people see it for triage.

See https://feedback.givewp.com/bug-reports/p/custom-fields-and-switching-payment-method-does-not-keep-data-in-custom-fields

**Comment by kjohnson** [comment]
Closing this Discussion in favor of the newly created Feedback.
